torch/_functorch/partitioners.py

```diff
@@ -580,9 +580,7 @@ def reordering_to_mimic_autograd_engine(gm: fx.GraphModule) -> fx.GraphModule:
     for node in gm.graph.find_nodes(op="placeholder"):
         env[node] = new_graph.node_copy(node, lambda x: env[x])
 
-    order = {}
-    for idx, node in enumerate(gm.graph.nodes):
-        order[node] = idx
+    order = {node: idx for idx, node in enumerate(gm.graph.nodes)}
```

Comment:
@Skylion007, it looks like this BE change is [causing flaky benchmark regression](https://hud.pytorch.org/hud/pytorch/pytorch/8bc7bd94a58c31507c4178a3fa64c86cbaf5a38e/1?per_page=50&name_filter=pull+%2F+cuda12.4-py3.10-gcc9-sm75+%2F+test+%28pr_time_benchmarks%2C+1%2C+1%2C+linux.g4dn.metal.nvidia.gpu%29&mergeLF=true).

Particularly, this line is sus, as it's in the codepath of the benchmark.

Reply:
ok, looks like benchmark is still failing, so this PR is unrelated, can be relanded safely